File: examples/typescript/list_check_types.ts
/**
 * Minimal end-to-end example for the Pescheck TypeScript client.
 *
 * Lists the check types supported by the API (GET /api/v2/checks/) and prints a
 * single line on success. Intended to be run in CI as a smoke test.
 *
 * Required environment variables:
 *   PESCHECK_BASE_URL       e.g. https://api-staging.pescheck.io
 *   PESCHECK_ACCESS_TOKEN   OAuth2 / JWT bearer access token
 */
import { Configuration, ChecksApi } from '@pescheckit/api-client';

async function main(): Promise<number> {
  const basePath = process.env.PESCHECK_BASE_URL;
  const accessToken = process.env.PESCHECK_ACCESS_TOKEN;
  if (!basePath || !accessToken) {
    console.error('ERROR: PESCHECK_BASE_URL and PESCHECK_ACCESS_TOKEN must be set');
    return 2;
  }

  // The accessToken config sets the OAuth2 / JWT bearer Authorization header.
  const config = new Configuration({ basePath, accessToken });
  const api = new ChecksApi(config);

  try {
    const checkTypes = await api.v2ChecksList();
    console.log(`E2E OK: ${checkTypes.length} check types`);
    return 0;
  } catch (error) {
    console.error(`ERROR: API call failed: ${error}`);
    return 1;
  }
}

main()
  .then((code) => process.exit(code))
  .catch((error) => {
    console.error(`ERROR: ${error}`);
    process.exit(1);
  });

File: examples/python/list_check_types.py
#!/usr/bin/env python3
"""Minimal end-to-end example for the Pescheck Python client.

Lists the check types supported by the API (GET /api/v2/checks/) and prints a
single line on success. Intended to be run in CI as a smoke test.

Required environment variables:
  PESCHECK_BASE_URL       e.g. https://api-staging.pescheck.io
  PESCHECK_ACCESS_TOKEN   OAuth2 / JWT bearer access token
"""
import os
import sys

import pescheck
from pescheck.rest import ApiException


def main() -> int:
    base_url = os.environ.get("PESCHECK_BASE_URL")
    token = os.environ.get("PESCHECK_ACCESS_TOKEN")
    if not base_url or not token:
        print(
            "ERROR: PESCHECK_BASE_URL and PESCHECK_ACCESS_TOKEN must be set",
            file=sys.stderr,
        )
        return 2

    configuration = pescheck.Configuration(host=base_url)
    # This API authenticates with an OAuth2 / JWT bearer access token.
    configuration.access_token = token

    try:
        with pescheck.ApiClient(configuration) as api_client:
            api_instance = pescheck.ChecksApi(api_client)
            check_types = api_instance.v2_checks_list()
    except ApiException as e:
        print(f"ERROR: API call failed: {e}", file=sys.stderr)
        return 1
    except Exception as e:  # noqa: BLE001 - surface any failure for CI
        print(f"ERROR: {e}", file=sys.stderr)
        return 1

    print(f"E2E OK: {len(check_types)} check types")
    return 0


if __name__ == "__main__":
    sys.exit(main())
